### PR TITLE
Fix selectLookupTest under VBlockTests

### PR DESCRIPTION
**Pull request by h-haddad**

### CODE REVIEW

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt [hunk 1]
```diff
@@ -91,21 +91,17 @@ class VBlockTests : JApplicationTestBase() {
...
 
     val vExecFailedException = assertFailsWith<VExecFailedException> {
       transaction {
-        FormWithList.block3.vBlock.refreshLookup(0)
+        FormWithList.block.vBlock.refreshLookup(0)
       }
     }
 
-    assertEquals("VIS-00016: Aucune valeur appropriée dans KOPI_USERS.", vExecFailedException.message)
+    assertEquals("VIS-00016: No matching value in MODULE.", vExecFailedException.message)
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt:104]: use `MessageCode.getMessage("VIS-00016", FormWithList.block.m.tableName)` for the exception message

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt [hunk 2]
```diff
@@ -91,21 +91,17 @@ class VBlockTests : JApplicationTestBase() {
   }
 
   @Test
-  @Ignore("TODO: this doesn't seem to be right")
   fun selectLookupTest() {
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt:94]: The name of this test should be renamed as it represents a fail scenario.
We should have another test that expects selectLookup to work successfully.

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithList.kt [hunk 3]
```diff
@@ -240,6 +240,12 @@ class BlockWithManyTables : FormBlock(20, 20, "Test block") {
     columns(r.module)
   }
 
+  val moduleName = visit(domain = Domain<String>(20), position = at(1, 1)) {
+    label = "module name"
+    help = "module name"
+    columns(m.shortName)
+  }
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithList.kt:247]: I think this field represents the shortName not the module name

**Review by hfazai — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt [hunk 4]
```diff
@@ -2964,10 +2964,12 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
 
       try {
         val result = table.slice(columns).select(conditions.compoundAnd()).single()
+        var j = 0
 
         fields.forEachIndexed { index, field ->
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt:2969]: `fields.forEach { field ->`

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt [hunk 5]
```diff
@@ -91,17 +92,29 @@ class VBlockTests : JApplicationTestBase() {
   }
 
   @Test
-  fun selectLookupTest() {
+  fun selectLookupTest_NoMatchingValueInTheTable() {
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt:95]: Could you rename the name of this test? you can use `` if a name is long.

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt [hunk 6]
```diff
@@ -91,17 +92,29 @@ class VBlockTests : JApplicationTestBase() {
...
         FormWithList.block.vBlock.refreshLookup(0)
       }
     }
 
-    assertEquals("VIS-00016: No matching value in MODULE.", vExecFailedException.message)
+    assertEquals(MessageCode.getMessage("VIS-00016", FormWithList.block.m.tableName), vExecFailedException.message)
+  }
+
+  @Test
+  fun selectLookupTest_Successful() {
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt:109]: Could you rename the name of this test? you can use `` if a name is long.

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithList.kt [hunk 7]
```diff
@@ -234,15 +234,9 @@ class BlockWithManyTables : FormBlock(20, 20, "Test block") {
     columns(m.id, r.module)
   }
 
-  val module = visit(domain = Domain<Int>(20), position = at(1, 1)) {
-    label = "module"
-    help = "module"
-    columns(r.module)
-  }
-
-  val moduleName = visit(domain = Domain<String>(20), position = at(1, 1)) {
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithList.kt]: Why you are removing this field?

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt [hunk 8]
```diff
@@ -91,21 +92,29 @@ class VBlockTests : JApplicationTestBase() {
   }
 
   @Test
-  @Ignore("TODO: this doesn't seem to be right")
-  fun selectLookupTest() {
+  fun selectLookupTest_NoMatchingValueInTheTable() {
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt:95]: I suggest the name:  `test VBlock.refreshLookup with no matching value in the table`

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt [hunk 9]
```diff
@@ -91,21 +92,29 @@ class VBlockTests : JApplicationTestBase() {
...
+        FormWithList.block.vBlock.refreshLookup(0)
       }
     }
 
-    assertEquals("VIS-00016: Aucune valeur appropriée dans KOPI_USERS.", vExecFailedException.message)
+    assertEquals(MessageCode.getMessage("VIS-00016", FormWithList.block.m.tableName), vExecFailedException.message)
+  }
+
+  @Test
+  fun selectLookupTest_Successful() {
```
Comment (hfazai) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VBlockTests.kt:109]: You can name this "refreshLookupTest"

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithList.kt (hunk 7 above)
Comment (h-haddad) [reply galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithList.kt]: Redundant field, we have mid that contain : columns(m.id, r.module)

**Review by hfazai — APPROVED** [review]

**Pull request merged**